fst/fst_query.py

```diff
@@ -3,7 +3,7 @@
 import re
 import time
 from pathlib import Path
-from watchdog.observers import Observer
+from watchdog.observers.polling import PollingObserver
```

Comment:
Can you try using the orginal implementation with WSL2?

The most performant observer implementation within the Watchdog library that is platform-agnostic is the **`watchdog.observers.Observer`** class. The **`Observer`** class is actually a platform-specific implementation that automatically selects the most suitable observer for the platform it is running on. When you import and use the **`Observer`** class, Watchdog internally uses the most efficient observer for your platform.

For example, on macOS and Linux, it will use the **`InotifyObserver`** or **`FSEventsObserver`**, while on Windows, it will use the **`WindowsApiObserver`**. These platform-specific observers are more performant than the **`PollingObserver`**, which relies on periodic polling of the file system.

In summary, using the **`watchdog.observers.Observer`** class is recommended, as it provides the most performant and platform-agnostic solution.

Yes, the Watchdog library should work with WSL (Windows Subsystem for Linux). When using WSL, the environment is similar to a native Linux environment. As a result, the watchdog.observers.Observer class should use the InotifyObserver for monitoring file system events.

However, it's important to note that WSL has certain limitations and differences compared to a native Linux environment. Some users have reported issues with file system events when using WSL, especially WSL1. WSL2, the newer version, has made significant improvements in compatibility and performance, so using WSL2 is recommended if you encounter any issues with file system event monitoring.

In summary, the Watchdog library should work with WSL, but it's better to use WSL2 for a more reliable experience.

If it does NOT work, I'll merge this PR!

Reply:
Okay that's helpful to know. I did try it on mac and didn't notice a big performance hit. I'll go ahead and merge, and we can easily revert when the time comes!